### PR TITLE
Writing flow: simplify & fix tabbing out of block

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

In #19235, I added logic for tabbing to move focus before or after the block list if focus leaves the block. This adjusts part of it to avoid searching tabbables and based on elements outside `WritingFlow`.

It also fixes Storybook. The block editor currently errors if you tab out of it.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

File: packages/block-editor/src/components/writing-flow/index.js [hunk 1]
```diff
@@ -77,13 +77,23 @@ export function isNavigationCandidate( element, keyCode, hasModifier ) {
  * Renders focus capturing areas to redirect focus to the selected block if not
  * in Navigation mode.
  */
-function FocusCapture( { selectedClientId, isReverse, containerRef } ) {
+const FocusCapture = forwardRef( ( {
```
Comment (youknowriad) [inline packages/block-editor/src/components/writing-flow/index.js]: Things are becoming a little bit hard to follow for me. it would be good to clarify exactly what this component is for, its props...

Comment (ellatrix) [reply packages/block-editor/src/components/writing-flow/index.js]: Sorry, I should be documenting better! I'll write something for it.

Comment (ellatrix) [reply packages/block-editor/src/components/writing-flow/index.js]: Added some in #19314.
